addons/mail/static/tests/discuss_tests.js

```diff
@@ -1068,5 +1068,84 @@ QUnit.test('mark all messages as read from Inbox', function (assert) {
     });
 });
 
+QUnit.test('reply to message from inbox', function (assert) {
+    var done = assert.async();
+    assert.expect(11);
+
+    var self = this;
+    this.data['mail.message'].records = [{
+        author_id: [5, 'Demo User'],
+        body: '<p>test 1</p>',
+        id: 1,
+        needaction: true,
+        needaction_partner_ids: [3],
+        res_id: 100,
+        model: 'some.document',
+        record_name: 'SomeDocument',
+    }];
+    this.data.initMessaging = {
+        needaction_inbox_counter: 1,
+    };
+
+    createDiscuss({
+        id: 1,
+        context: {},
+        params: {},
+        data: this.data,
+        services: this.services,
+        session: { partner_id: 3 },
+        mockRPC: function (route, args) {
+            if (args.method === 'message_post') {
+                assert.step(args.method);
+                assert.strictEqual(args.model, 'some.document',
+                    "should post message to correct document model");
+                assert.strictEqual(args.args[0], 100,
+                    "should post message to correct document ID");
+
+                self.data['mail.message'].records.push({
```

Comment:
is this really necessary?

Reply:
Yes, otherwise it crashes: https://github.com/odoo/odoo/blob/12.0/addons/mail/static/src/js/models/threads/document_thread.js#L333

Reply:
then this should probably be done in mockserver at some point